src/test/java/com/redhat/pantheon/servlet/DefaultQueryServletTest.java

```diff
@@ -0,0 +1,203 @@
+
+    @Test
+    @DisplayName("Test a simple query")
+    public void testSimpleQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
+
+        // When
+        servlet.doGet(slingContext.request(), slingContext.response());
+
+        // Then
+        JsonNode jsonNode = new ObjectMapper().readTree(slingContext.response().getOutputAsString());
+        assertEquals(10L, jsonNode.get("size").asLong());
+        assertEquals(10L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
+    }
+
+    @Test
+    @DisplayName("Test a sorted query")
+    public void testSortedQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
+        slingContext.request().getParameterMap().put("orderBy", new String[]{"name desc"});
+
+        // When
+        servlet.doGet(slingContext.request(), slingContext.response());
+
+        // Then
+        JsonNode jsonNode = new ObjectMapper().readTree(slingContext.response().getOutputAsString());
+        assertEquals(10L, jsonNode.get("size").asLong());
+        assertEquals(10L, jsonNode.get("nextOffset").asLong());
+        assertTrue(jsonNode.get("results").isArray());
+        Iterator<JsonNode> resultIterator = jsonNode.get("results").iterator();
+        // verify the order is inversed
+        for (int i = 0; i < 10; i++) {
+            assertEquals("node-" + (9-i), resultIterator.next().get("name").asText());
+        }
+    }
+
+    @Test
+    @DisplayName("Test a filtered query")
+    public void testFilteredQuery() throws Exception {
+        // Given
+        ResourceBuilder builder = slingContext.build();
+        for (int i = 0; i < 10; i++) {
+            builder.resource("/content/test/node-" + i,
+                    "name", "node-" + i,
+                    "number", i);
+        }
+        builder.commit();
+
+        DefaultQueryServlet servlet = new DefaultQueryServlet();
+        servlet.init();
+        slingContext.request().setResource(slingContext.resourceResolver().getResource("/content/test"));
```

Comment:
We seem to be repeating the above dozen lines; is it worth it to pull it out into some kind of testSetup method?

Reply:
Yes it is. I'll fix this.

Reply:
I will take out the common JCR repository prep code, but I'll leave some individual lines just to keep the tests independent.